src/components/date-range/date-range.component.js

```diff
@@ -34,89 +34,55 @@ const DateRange = ({
 
-  /** The startDate value */
   const getStartDate = useCallback(() => {
     const { value: startValue } = startDateProps;
-    const computedValue = isControlled ? value : defaultValue;
 
-    return startValue || computedValue ? computedValue[0] : undefined;
-  }, [defaultValue, isControlled, startDateProps, value]);
+    return startValue || value[0];
+  }, [startDateProps, value]);
 
-  /** The endDate value */
   const getEndDate = useCallback(() => {
     const { value: endValue } = endDateProps;
-    const computedValue = isControlled ? value : defaultValue;
 
-    return endValue || computedValue ? computedValue[1] : undefined;
-  }, [defaultValue, isControlled, endDateProps, value]);
+    return endValue || value[1];
+  }, [endDateProps, value]);
+
+  const [inputRefMap, setInputRefMap] = useState({
+    start: {
+      isBlurBlocked: { current: false },
+      setOpen: null,
+    },
+    end: {
+      isBlurBlocked: { current: false },
+      setOpen: null,
+    },
+  });
+
+  function isEmptyValue(allowEmpty, inputValue) {
+    return allowEmpty && !inputValue.length;
+  }
 
   const [startDateValue, setStartDateValue] = useState({
-    formattedValue: DateHelper.formatDateToCurrentLocale({
-      value: getStartDate(),
-      ...localeData,
-    }),
-    rawValue: DateHelper.formatValue({
-      value: getStartDate() || (startDateProps.allowEmptyValue ? "" : today),
-      ...localeData,
-    }),
+    formattedValue: getStartDate(),
+    rawValue: isEmptyValue(startDateProps.allowEmptyValue, getStartDate())
+      ? ""
+      : formatToISO(format, getStartDate()),
   });
 
   const [endDateValue, setEndDateValue] = useState({
-    formattedValue: DateHelper.formatDateToCurrentLocale({
-      value: getEndDate(),
-      ...localeData,
-    }),
-    rawValue: DateHelper.formatValue({
-      value: getEndDate() || (endDateProps.allowEmptyValue ? "" : today),
-      ...localeData,
-    }),
+    formattedValue: getEndDate(),
+    rawValue: isEmptyValue(endDateProps.allowEmptyValue, getEndDate())
+      ? ""
+      : formatToISO(format, getEndDate()),
   });
 
-  const updateValues = useCallback(() => {
-    setStartDateValue({
-      formattedValue: DateHelper.formatDateToCurrentLocale({
-        value: getStartDate(),
-        ...localeData,
-      }),
-      rawValue: DateHelper.formatValue({
-        value: getStartDate() || (startDateProps.allowEmptyValue ? "" : today),
-        ...localeData,
-      }),
-    });
-
-    setEndDateValue({
-      formattedValue: DateHelper.formatDateToCurrentLocale({
-        value: getEndDate(),
-        ...localeData,
-      }),
-      rawValue: DateHelper.formatValue({
-        value: getEndDate() || (endDateProps.allowEmptyValue ? "" : today),
-        ...localeData,
-      }),
-    });
-  }, [
-    getEndDate,
-    getStartDate,
-    endDateProps,
-    startDateProps,
-    localeData,
-    today,
-  ]);
-
   function usePrevious(arg) {
```

Comment:
Maybe not in this PR, but I think this should be made into an internal hook, as the exact same one is used in Decimal as well

Reply:
Yeah agree, I actually have a PR for Drawer that adds as well 😄

Reply:
Yeah I agree. It's used in a couple of places now so probably warrants it's own custom hook 🪝